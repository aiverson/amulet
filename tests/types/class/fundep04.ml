class collects 'c 'e | 'c -> 'e begin
  val empty  : 'c
  val insert : 'c -> 'e -> 'c
end

let ins2 xs a b = insert (insert xs a) b
