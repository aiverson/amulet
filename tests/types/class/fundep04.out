collects : type -> type -> constraint
empty : Spec{'c : type}. Spec{'e : type}. collects 'c 'e => 'c
insert : Spec{'c : type}. Spec{'e : type}. collects 'c 'e => 'c -> 'e -> 'c
ins2 : Infer{'a : type}. Infer{'b : type}. collects 'a 'b => 'a -> 'b -> 'b -> 'a
